purge: updated flow to allow the user to opt out

pcuzner opened the pull request:
Playbook performs some initial local checks, then checks
each ceph host has cephadm and the right fsid. Once
these plays are passed, the user is prompted to purge or
not, so they may abort

Closes #21, closes #22, closes #23

Signed-off-by: Paul Cuzner <pcuzner@redhat.com>

pcuzner committed:
purge: updated flow to allow the user to opt out

Playbook performs some initial local checks, then checks
each ceph host has cephadm and the right fsid. Once
these plays are passed, the user is prompted to purge or
not, so they may abort

Closes #21, closes #22, closes #23

Signed-off-by: Paul Cuzner <pcuzner@redhat.com>

guits commented on cephadm-purge-cluster.yml:
```diff
@@ -45,16 +43,79 @@
+      fail:
+        msg: >
+          The fsid directory '/var/lib/ceph/{{ fsid }}' can not be found on {{ inventory_hostname }}
+      when: fsid_exists.stat.exists == false and group_names != ['clients']
```
```suggestion
      when:
        - not fsid_exists.stat.exists | bool
        - group_names != ['clients']
```

pcuzner replied:
ACK

guits commented on cephadm-purge-cluster.yml:
```diff
@@ -45,16 +43,79 @@
+        msg: |
+          The cephadm binary is missing on {{ inventory_hostname }}. To purge the cluster you must have cephadm installed
+          on ALL ceph hosts
+      when: cephadm_exists.rc not in [ 0 ] and group_names != ['clients']
```
```suggestion
      when:
        - cephadm_exists.rc not in [ 0 ]
        - group_names != ['clients']
```

pcuzner replied:
ACK

pcuzner commented:
changes applied, commits squashed

guits commented:
jenkins test all_daemons

guits commented:
jenkins test deploy_and_purge

guits commented:
jenkins test deploy_and_purge

guits commented:
jenkins test deploy_and_purge

guits commented on cephadm-purge-cluster.yml:
```diff
@@ -11,29 +11,27 @@
           admin keyring is present at /etc/ceph/admin.client.keyring
       when: "'admin' not in groups or groups['admin'] | length < 1"
 
-- name: make sure admin keyring is present and pause cephadm
```
why did you remove this? (L32-L36)
the presence of this keyring must be checked on the node admin[0]

pcuzner replied:
a sugar low..no idea. thanks for catching. The check is in the wrong place - I had it in the localhost checks!

pcuzner committed:
purge: update tox to remove the test cluster

Now that the purge playbook has an opt out, tox
needed to change to allow the purge to run

Signed-off-by: Paul Cuzner <pcuzner@redhat.com>

guits commented:
@pcuzner could you squash the two first commits please?

thanks!

pcuzner commented:
> @pcuzner could you squash the two first commits please?
> 
> thanks!

squashed - so you have one commit handling the playbook changes and another tox - hope that's what you meant

guits commented:
> > @pcuzner could you squash the two first commits please?
> > thanks!
> 
> squashed - so you have one commit handling the playbook changes and another tox - hope that's what you meant

yup :slightly_smiling_face:
